### PR TITLE
Feature auth

### PR DESCRIPTION
**Pull request by tuj**
* Added token authentication based on allowed agencies.
* Added behat tests.
* Upgrade to latest symfony.
* Downgrade phpunit to 7

### CODE REVIEW

File: src/Security/TokenAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    'auth_basic' => [$this->clientId, $this->clientSecret],
+                ]
+            );
+
+            if (200 !== $response->getStatusCode()) {
+                return null;
+            }
+
+            $content = $response->getContent();
+            $data = json_decode($content);
```
Comment (cableman) [inline src/Security/TokenAuthenticator.php]: Maybe we should add check that the response is json (json_decode) may decode to FALSE if not valid json and then `false === $data->active` would faild? We should handle this bette and log information for debugging about this

Comment (cableman) [reply src/Security/TokenAuthenticator.php]: Yes I known it comes from the code I made in upload :-) sorry

File: src/Security/TokenAuthenticator.php [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        } catch (ExceptionInterface $e) {
+            return null;
+        }
+
+        // Create user object.
+        $tokenExpireDataTime = new \DateTime($data->expires, new \DateTimeZone('Europe/Copenhagen'));
+        $user = new User();
+        $user->setPassword($token);
+        $user->setExpires($tokenExpireDataTime);
+        $user->setAgency($data->agency);
```
Comment (cableman) [inline src/Security/TokenAuthenticator.php]: If this was not an agency token but an user token we need to detected that as I don't think an user token has agency on it. So this `$data->agancy` would result in a 500 error.

We may need to look at detecting token type an give an invalided token or access denied (return null)

File: src/Security/TokenAuthenticator.php [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                return null;
+            }
+        } catch (HttpExceptionInterface $e) {
+            return null;
+        } catch (ExceptionInterface $e) {
+            return null;
+        }
+
+        // Create user object.
+        $tokenExpireDataTime = new \DateTime($data->expires, new \DateTimeZone('Europe/Copenhagen'));
```
Comment (cableman) [inline src/Security/TokenAuthenticator.php]: What if the token has expired? We should also tjek for that?

**Review by cableman — CHANGES_REQUESTED** [review]
These comments need to be back-ported to frontend and upload auth as well

File: src/Security/TokenAuthenticator.php (hunk 1 above)
Comment (tuj) [reply src/Security/TokenAuthenticator.php]: Fixed

File: src/Security/TokenAuthenticator.php (hunk 3 above)
Comment (tuj) [reply src/Security/TokenAuthenticator.php]: Added expires checks

File: src/Security/TokenAuthenticator.php (hunk 2 above)
Comment (tuj) [reply src/Security/TokenAuthenticator.php]: Fixed

**Review by turegjorup — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Pull request merged**